### DISCUSSION TITLE
trigger update_sa with destination address change

### DISCUSSION THREAD

**Discussion by deeptishm** [strongswan/strongswan#1207] [Q&A]
Hi,
Under what conditions can the destination ip change be simulated for update_sa?
Using https://www.strongswan.org/testing/testresults/ikev2/mobike-virtual-ip-nat/index.html, we could simulate change in src ip; however we need to test change in destination ip address.

Many thanks in advance.

Thanks.

**Comment by Thermi** [comment]
what do you mean? whose IP address exactly?

**Reply by strongX509** [reply]
MOBIKE only can detect changes in the source IP address. The only way a change in the destination  IP address can be detected is by running MOBIKE on the peer.

**Reply by deeptishm** [reply]
Thanks for your replies. Is there a referenec setup and configuration similar to https://www.strongswan.org/testing/testresults/ikev2/mobike-virtual-ip-nat/index.html that i can use?

**Comment by strongX509** [comment]
Well just enable MOBIKE on the destination. `mobike = yes` is the default anyway. The problem is a NAT router near the source. After a destination IP address change, the destination might not be able to reach the source with IKEv2 INFORMATIONAL messages since there will be no matching entry in the mapping table of the NAT router. In that case only DPD with `restart` on the source and `clear` on the destination side will help. Unfortunately the DPD mechanism takes a couple of minutes until a connection break is detected and a new IPsec tunnel is established. And the new destination address must of course be resolvable via dynamic DNS.

**Reply by deeptishm** [reply]
Thanks will try this out.
